### ISSUE TITLE
Strange "reject" error

### ISSUE THREAD

**Issue by thelinuxlich** [neumino/rethinkdbdash#99]
I'm getting a error "TypeError: Cannot read property 'reject' of undefined":

```
  File "/app/releases/20150624035113/node_modules/rethinkdbdash/lib/connection.js", line 170, in Connection._processResponse
    if (typeof self.metadata[token].reject === 'function') {
  File "/app/releases/20150624035113/node_modules/rethinkdbdash/lib/connection.js", line 129, in Socket.<anonymous>
    self._processResponse(response, token);
  File "events.js", line 77, in emitOne
  File "events.js", line 166, in Socket.emit
  File "_stream_readable.js", line 145, in readableAddChunk
  File "_stream_readable.js", line 109, in Socket.Readable.push
  File "net.js", line 509, in TCP.onread
```

**Comment by neumino** [comment]
What the query that trigger this error?

A readable stream? Is it on a table? Or on a changefeed?

**Comment by thelinuxlich** [comment]
It is on a table, I'm gonna find out what query triggered the error

**Comment by neumino** [comment]
@thelinuxlich -- did you close a cursor while fetching data from it maybe?

**Comment by neumino** [comment]
Ok, I found a case when this could happen.
If you have a cursor on a query, and call `cursor.stop` since the protocol `v0_4` we will send a STOP query. At that time we have to outgoing request.

In your stacktrace, we are getting a `CLIENT_ERROR`. My guess is that the cursor returned a `CLIENT_ERROR` for the `CONTINUE` query, we then deleted the metadata, and we get a second time a `CLIENT_ERROR` for the `STOP` query - and at that time we get your error.

It should be fixed in 2.0.15.
Feel free to reopen if it happens again :)

**Comment by thelinuxlich** [comment]
I will update, but I wasn't using cursors either
